crypto/src/swap.rs

```diff
@@ -26,10 +56,129 @@ pub struct SwapPlaintext {
 }
 
 impl SwapPlaintext {
-    // TODO: needs to be constant-length
-    pub fn encrypt(&self, _ovk: &OutgoingViewingKey) -> SwapCiphertext {
-        // TODO: implement
-        todo!()
+    // Create a new hash based on the ephemeral public key and shared secret suitable for use as a key for symmetric encryption.
+    //
+    // Implementing this way allows recovery of all swap plaintexts via the seed phrase.
+    //
+    // Theoretically, if a paranoid user did want to achieve forward secrecy, they could choose to encrypt
+    // nonsense bytes as the swap plaintext as the swap ciphertext does not need to be valid for the
+    // swap to succeed, however this is unsupported by the official client.
+    fn derive_symmetric_key(
+        shared_secret: &ka::SharedSecret,
+        epk: &ka::Public,
+    ) -> blake2b_simd::Hash {
+        let mut kdf_params = blake2b_simd::Params::new();
+        kdf_params.hash_length(32);
+        let mut kdf = kdf_params.to_state();
+        kdf.update(&shared_secret.0);
+        kdf.update(&epk.0);
+
+        kdf.finalize()
+    }
```

Comment:
Can we share this code with the shared secret derivation used for encrypting the note plaintext that records the swap nft?

Reply:
I think you can import `note::derive_symmetric_key` instead of adding it here (we also import `note::derive_symmetric_key` when doing the memo encryption)